sap_hana/datadog_checks/sap_hana/queries.py

```diff
@@ -0,0 +1,342 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+
+"""
+https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.02/en-US/20cbb10c75191014b47ba845bfe499fe.html
+
+> The views in the SYS_DATABASES schema provide aggregated information from a subset of the views available in the
+  SYS schema of all tenant databases in the system. These union views have the additional column DATABASE_NAME to
+  allow you to identify to which database the information refers.
+
+To find out exactly what is available in each subset, run:
+
+ALL_SCHEMA = 'SYS_DATABASES'
+_ = cur.execute("SELECT VIEW_NAME FROM VIEWS WHERE SCHEMA_NAME = '{ALL_SCHEMA}'")
+views = [row[0] for row in sorted(cur.fetchall())]
+for view in views:
+    print(view)
+    _ = cur.execute(f"SELECT COLUMN_NAME FROM VIEW_COLUMNS WHERE SCHEMA_NAME = '{ALL_SCHEMA}' and VIEW_NAME = '{view}'")
+    for row in cur.fetchall():
+        print(f'    {row[0]}')
+"""
+import inspect
+from collections import defaultdict
+
+from .utils import compact_query
+
+VIEWS_USED = defaultdict(set)
+
+
+def view(v, user=None):
+    VIEWS_USED[user or inspect.getframeinfo(inspect.currentframe().f_back).function].add(v)
```

Comment:
I think I see the point of that method, but I'm not sure it's better than maintaining a list of views, considering we still need to explicitly implement the support for the view in the check anyway.

Reply:
I'm a bit confused around the code around the `views`, seems a bit too clever :) I think there is probably a better way.